RocketChat/Rocket.Chat
# fix: Message count is not changing after removing channel selection

## Proposed changes (including videos or screenshots)


* Created a new hook to filter messages with the corresponding userId or roomId from the channel.
* Created a new `messages` prop on the IImportFileData.ts: `messages: Array<Pick<IImportMessage, '_id' | 'rid' | 'u'>>;`



## Issue(s)

WM-123
## Steps to test or reproduce

*Steps to reproduce:*

1. Open RC
2. Click on “Workspace” → Import
3. Click on “Import new file”
4. Select the import optione.g: Slack
5. Upload the file
6. In the table, remove some channels from the selection

*Expected Behavior:*
The number of messages should change after removing some channels (with messages) from the selection

*Actual Behavior:*
The number is fixed

![image](https://github.com/RocketChat/Rocket.Chat/assets/84182706/3479b9ef-02f1-49b7-9b36-6e67711cac3b)


## Further comments
